app/consumers/compliance_reports_consumer.rb

```diff
@@ -1,16 +1,20 @@
 # frozen_string_literal: true
 
+# Raise an error if entitlement is not available
+class EntitlementError < StandardError; end
+
 # Receives messages from the Kafka topic, converts them into jobs
 # for processing
 class ComplianceReportsConsumer < ApplicationConsumer
   subscribes_to Settings.platform_kafka_topic
 
   def process(message)
     @msg_value = JSON.parse(message.value)
-    logger.info "Received message, enqueueing: #{message.value}"
+    raise EntitlementError unless identity.valid?
```

Comment:
We could also do something like `identity.validate!` which would raise this error, and that would keep the error defined within the `IdentityHeader` class. (optional)